Top-level docs rewrite

errordeveloper opened the pull request:
<!-- Reviewable:start -->
This change is [<img src="https://reviewable.io/review_button.svg" height="35" align="absmiddle" alt="Reviewable"/>](https://reviewable.io/reviews/weaveworks/weave-kubernetes-anywhere/56)
<!-- Reviewable:end -->

errordeveloper commented:
@abuehrle please take a look when you have a moment, this incorporates your changes to top-level docs fro from #30.

peterbourgon commented on README.md:
```diff
@@ -1,18 +1,57 @@
+
+# What is it made of?
+
+We are currently shipping descrete containers (no self-hosting). You will find up-to-date images on Docker
```
discrete

peterbourgon commented on README.md:
```diff
@@ -1,18 +1,57 @@
+
+These images are based on upstream `gcr.io/google_containers/hyperkube` and incorporate thin shell scripts that
+pass correct arguments to each of the components. You may wish to explore [`docker-images`][docker-images]
+directory, if you with to know what goes in exactly.
```
**want** to know

peterbourgon commented on README.md:
```diff
@@ -1,18 +1,57 @@
+directory, if you with to know what goes in exactly.
+
+Images are tagged with component name and version. If you pull `weaveworks/kubernetes-anywhere:apiserver-v1.2`,
+you will get the latest build of `v1.2.x`, so you upgrading your cluster to patch releases is very easy.
```
so ~~you~~ upgrading your cluster

peterbourgon commented on README.md:
```diff
@@ -1,18 +1,57 @@
+Images are tagged with component name and version. If you pull `weaveworks/kubernetes-anywhere:apiserver-v1.2`,
+you will get the latest build of `v1.2.x`, so you upgrading your cluster to patch releases is very easy.
+However, if you don't like this, there are tags for patch releases also. We may update any of the versioned
+image without changing the version of Kubernetes base image, but you can use Docker's new content hashes,
```
We may update any of the versioned **images**

peterbourgon commented on README.md:
```diff
@@ -29,9 +68,24 @@ module "kubernetes-anywhere-aws-ec2" {
+one master and 3 workers.  All resources will be tagged `KubernetesCluster=weave1`.
+The module code can be found in [`examples/aws-ec2-terraform`][aws-ec2-terraform].
+
+Once instances are up, you can login to either of the 3 worker nodes or the master and run 
```
login to **any** of the 3 worker nodes

peterbourgon commented on README.md:
```diff
@@ -315,20 +372,19 @@ service "kube-dns" created
-[`weaveworks/kubernetes-anywhere`][docker-hub] are kept up-to-date, however there may be a good reason for the user to rebuild
-these images and publish in their own registry, please see [`docker-images`][docker-images] directory for scripts
-and `Dockerfile`s used to build the images.
+These examples are designed to be easy to adopt and therefor are kept simple. Your production requirement may vary,
```
therefore

peterbourgon replied:
production **requirements**

peterbourgon commented on README.md:
```diff
@@ -315,20 +372,19 @@ service "kube-dns" created
-these images and publish in their own registry, please see [`docker-images`][docker-images] directory for scripts
-and `Dockerfile`s used to build the images.
+These examples are designed to be easy to adopt and therefor are kept simple. Your production requirement may vary,
+catering for all would make these example too complex.
```
make these **examples**

errordeveloper commented:
Thanks @peterbourgon!

rade commented on README.md:
```diff
@@ -1,18 +1,58 @@
+> [are provided](#enough-said--go-try-it).
 
-Additionally, thanks to how [Weave Net handles IP address allocation as well as DNS](https://www.youtube.com/watch?v=117gWVShcGU) without requiring a persistent store, you can deploy etcd over Weave Net as well. The etcd cluster can thereby benefit from simple service discovery WeaveDNS provides and therefore facilitate node replacement without config changes.
+Kubernetes Anywhere uses Weave Net to dramatically simplify Kubernetes deployment — anywhere.
```
I think Weave Net should be mentioned a little later. I.e. try listing the benefits of k8s-anywhere w/o mentioning Net.

rade replied:
by the same token, I think we should drop the "weave" from the project name. It may lead folks to erroneously assume that this is aimed at existing weave users.

errordeveloper replied:
I've renamed the repo. I agree that the first paragraph can definitely be improved and I'd like someone else to take a stab at it, but I don't see a reason to block on this.

rade commented on README.md:
```diff
@@ -1,18 +1,58 @@
-# Try it!
+[bryans-talk]: https://www.youtube.com/watch?v=117gWVShcGU
+
+# What is it made of?
```
Do we really need this section? Here? It tells me stuff w/o context. Why would I care that we ship discrete containers? Why would I want to build them myself? etc. All sounds like stuff for advanced users, hence should come later, and needs explaining.

errordeveloper replied:
Ok, I certainly must address this before merging.

rade commented on README.md:
```diff
@@ -1,18 +1,58 @@
 All you need is one or more Docker hosts.
 
-## Using Terraform in EC2
+## Full-blow Cluster Example (Using Terraform in EC2)
```
I think we should lead with the single-host docker example.

abuehrle commented:
I can't really effectively edit something using this inline tool in git. It's ok for typos, but for anything more substantial,  I am going to have to edit the file directly. 

Typically, I run the sample on my own machine to ensure it works and test out the instructions. This is the only way to see if the 'workflow' works, and to be able to make structural suggestions.  Hope that's ok.

abuehrle commented:
Questions: 

What version of the following should we be using:

-docker
-kubernetes
-weave net

Can an individual set this up without using the provided scripts?  How do we achieve that?

errordeveloper commented:
> What version of the following should we be using:
> - docker

Kubernetes Anywhere requires Docker 1.10 (current). I am testing 1.11 releases candidate right now, but we are still far from the final release, the `-rc1` has only been out for some number of hourse now...

> - kubernetes

Kubernetes Anywhere provides v1.1 and v1.2 right now.

> - weave net

Latest, i.e. 1.4.6. We will switch to 1.5 as soon as it's released.

> Can an individual set this up without using the provided scripts? How do we achieve that?

There are no external scripts as such, it's just Docker images. These images contains numerous small scripts which glue it all together, it would take forever to explain what each does and and it all fits in. There would be ifs all over the place. I don't see a point of breaking it up, the generalised multi-node instructions are already the most general thing we can have, although still quite verbose due to the nature of the beast...

errordeveloper commented:
*[README.md, line 383 \[r2\]](https://reviewable.io:443/reviews/weaveworks/kubernetes-anywhere/56#-KD_vwN8AJuTAwEhG1eF-r2-383:-KEGYp3BoaD12fWfLetY:-499146009) ([raw file](https://github.com/weaveworks/kubernetes-anywhere/blob/b634f42e7c98945b094d3ce45e941a71824a789f/README.md#L383)):*
Done.

---


*Comments from the [review on Reviewable.io](https://reviewable.io:443/reviews/weaveworks/kubernetes-anywhere/56)*
<!-- Sent from Reviewable.io -->

errordeveloper commented:
Review status: 0 of 1 files reviewed at latest revision, 10 unresolved discussions.

---

*[README.md, line 81 \[r2\]](https://reviewable.io:443/reviews/weaveworks/kubernetes-anywhere/56#-KD_vwN8AJuTAwEhG1eF-r2-81:-KEGYmoefdsAQ0jR3k-0:-499146009) ([raw file](https://github.com/weaveworks/kubernetes-anywhere/blob/b634f42e7c98945b094d3ce45e941a71824a789f/README.md#L81)):*
Done.

---

*[README.md, line 382 \[r2\]](https://reviewable.io:443/reviews/weaveworks/kubernetes-anywhere/56#-KD_vwN8AJuTAwEhG1eF-r2-382:-KEGYoMI6QpkvVEZFHDw:-499146009) ([raw file](https://github.com/weaveworks/kubernetes-anywhere/blob/b634f42e7c98945b094d3ce45e941a71824a789f/README.md#L382)):*
Done.

---

*[README.md, line 53 \[r3\]](https://reviewable.io:443/reviews/weaveworks/kubernetes-anywhere/56#-KD_vwN8AJuTAwEhG1eF-r3-53:-KEGYm3OQPS1-Cbgvh_S:-499146009) ([raw file](https://github.com/weaveworks/kubernetes-anywhere/blob/746cef52874fb1f6fb57a2d2f30b46bf15e66d09/README.md#L53)):*
Done.

---


*Comments from the [review on Reviewable.io](https://reviewable.io:443/reviews/weaveworks/kubernetes-anywhere/56)*
<!-- Sent from Reviewable.io -->

errordeveloper commented:
Review status: 0 of 1 files reviewed at latest revision, 10 unresolved discussions.

---

*[README.md, line 28 \[r2\]](https://reviewable.io:443/reviews/weaveworks/kubernetes-anywhere/56#-KD_vwN8AJuTAwEhG1eF-r2-28:-KEGZULnvWysBaXIeVt0:-499146009) ([raw file](https://github.com/weaveworks/kubernetes-anywhere/blob/b634f42e7c98945b094d3ce45e941a71824a789f/README.md#L28)):*
Done.

---

*[README.md, line 34 \[r2\]](https://reviewable.io:443/reviews/weaveworks/kubernetes-anywhere/56#-KD_vwN8AJuTAwEhG1eF-r2-34:-KEGZS1L5sL_DJ4mGgp7:-499146009) ([raw file](https://github.com/weaveworks/kubernetes-anywhere/blob/b634f42e7c98945b094d3ce45e941a71824a789f/README.md#L34)):*
Done.

---

*[README.md, line 37 \[r2\]](https://reviewable.io:443/reviews/weaveworks/kubernetes-anywhere/56#-KD_vwN8AJuTAwEhG1eF-r2-37:-KEGZRdIIs0Ul9QmQLyM:-499146009) ([raw file](https://github.com/weaveworks/kubernetes-anywhere/blob/b634f42e7c98945b094d3ce45e941a71824a789f/README.md#L37)):*
Done.

---

*[README.md, line 39 \[r2\]](https://reviewable.io:443/reviews/weaveworks/kubernetes-anywhere/56#-KD_vwN8AJuTAwEhG1eF-r2-39:-KEGZQv-J4oZ4ktr_ZCL:-499146009) ([raw file](https://github.com/weaveworks/kubernetes-anywhere/blob/b634f42e7c98945b094d3ce45e941a71824a789f/README.md#L39)):*
Done.

---


*Comments from the [review on Reviewable.io](https://reviewable.io:443/reviews/weaveworks/kubernetes-anywhere/56)*
<!-- Sent from Reviewable.io -->

errordeveloper commented:
I'm leaning towards merging this as is, it's already so much better then the current readme...

errordeveloper commented:
Also, from conversation with @peterbourgon I should improve the note about Docker setup.

1. [ ] move the note about Docker setup into a separate section
2. [ ] ensure that version of Docker is specified more explicitly
3. [ ] state explictitly that local Docker is required (#54)
4. [ ] describe the issue with `docker.service` linking to relevant issue in Docker repo and stating explicitly that the instructions related to the host and not the toolbox container or anything
5. [ ] add a note about Debian 8 issue (https://github.com/kubernetes/kubernetes/issues/23816)

errordeveloper commented:
also, cc #55

errordeveloper commented:
I've addressed most of the concerns which resulted in quite a few changes. I'm going to merge it now, on the basis that it's better then the current version, so please file issues for any further improvements you may want to discuss. // cc @rade @peterbourgon @abuehrle

errordeveloper commented:
*[README.md, line 27 \[r3\]](https://reviewable.io:443/reviews/weaveworks/kubernetes-anywhere/56#-KD_vwN8AJuTAwEhG1eF-r3-27:-KGRU9ZE_EWNvOzf8cce:-499146009) ([raw file](https://github.com/weaveworks/kubernetes-anywhere/blob/746cef52874fb1f6fb57a2d2f30b46bf15e66d09/README.md#L27)):*
Done.

---


*Comments from [Reviewable](https://reviewable.io:443/reviews/weaveworks/kubernetes-anywhere/56)*
<!-- Sent from Reviewable.io -->

errordeveloper commented:
Reviewed 2 of 2 files at r7.
Review status: all files reviewed at latest revision, 10 unresolved discussions, some commit checks failed.

---



*Comments from [Reviewable](https://reviewable.io:443/reviews/weaveworks/kubernetes-anywhere/56)*
<!-- Sent from Reviewable.io -->

errordeveloper commented:
Reviewed 2 of 2 files at r7.
Review status: all files reviewed at latest revision, 10 unresolved discussions.

---



*Comments from [Reviewable](https://reviewable.io:443/reviews/weaveworks/kubernetes-anywhere/56)*
<!-- Sent from Reviewable.io -->
